nixos/modules/config/users-groups.nix

```
@@ -578,7 +581,8 @@ in {
         # there is at least one "privileged" account that has a
         # password or an SSH authorized key. Privileged accounts are
         # root and users in the wheel group.
-        assertion = !cfg.mutableUsers ->
+        # Both allowLoginless and mutableUsers should be false to trigger the check.
+        assertion = !(cfg.mutableUsers || cfg.allowNoPasswordLogin) ->
```

```suggestion
        # The check does not apply when users.disableLoginPossibilityAssertion
        # The check does not apply when users.mutableUsers
        assertion = !cfg.mutableUsers -> !cfg.allowNoPasswordLogin ->
```

I don't feel particularly strongly about the refactor, but the comment should be clearer.